pygeoapi_plugins/provider/geopandas_provider.py

```diff
@@ -0,0 +1,492 @@
+        self._set_geometry_fields(provider_def)
+            
+        self.gdf[self.id_field] = self.gdf[self.id_field].astype(str)
+        
+        # Without below, the CSV reads std_id as an object dtype
+        # And fails the CSV provider tests. Maybe a way to do this better
+        # that is more generalizable?
+        if "stn_id" in self.gdf.columns:
+            self.gdf["stn_id"] = self.gdf["stn_id"].astype('int64')
+        if "value" in self.gdf.columns:
+            self.gdf["value"] = self.gdf["value"].astype('float64')
+
+        self._exclude_from_properties: list[str] = self._exclude_from_fields + [self.id_field]
+
+        self.fields = self.get_fields()
+
+        
+    def get_fields(self) -> dict[str, any]:
+        """
+        Get provider field information (names, types)
+
+        Example response: {'field1': 'string', 'field2': 'number'}}
+
+        :returns: dict of field names and their associated JSON Schema types
+        """
+        if len(self.gdf) == 0:
+            raise ProviderNoDataError("No data found to get fields from")
+        
+
+        field_mapper =  {col: self.gdf[col].dtype.name for col in self.gdf.columns 
+                         if col not in self._exclude_from_fields
+                         }
+        
+       # Pandas has a different ãmes for types than the pygeoapi spec expects
+        pandas_dtypes_to_ours = {
+            'float64': 'number',
+            'int64': 'integer',
+            'object': 'string'
+        }
+
+        pandas_default = defaultdict(lambda: 'string')
+        pandas_default.update(pandas_dtypes_to_ours)
+
+        
+        our_types_names = {k: {
+            "type": pandas_default[v]
+            }
+        for k, v in field_mapper.items()
+        }
+
+        return our_types_names
+
+    def get_schema(self, schema_type: SchemaType = SchemaType.item):
+        """
+        Get provider schema model
+
+        :param schema_type: `SchemaType` of schema (default is 'item')
+
+        :returns: tuple pair of `str` of media type and `dict` of schema
+                  (i.e. JSON Schema)
+        """
+
+        raise NotImplementedError()
+        
+
+    def getgdf_path(self, baseurl, urlpath, dirpath):
+        """
+        Gets directory listing or file description or raw file dump
+
+        :param baseurl: base URL of endpoint
+        :param urlpath: base path of URL
+        :param dirpath: directory basepath (equivalent of URL)
+
+        :returns: `dict` of file listing or `dict` of GeoJSON item or raw file
+        """
+
+        raise NotImplementedError()
+
+    def get_metadata(self):
+        """
+        Provide data/file metadata
+
+        :returns: `dict` of metadata construct (format
+                  determined by provider/standard)
+        """
+
+        raise NotImplementedError()
+
+    # we want to support bbox and datetime_
+    def query(self, offset=0, limit=10, resulttype: Literal['results', 'hits']='results',
```

Comment:
add **kwargs to arguments of this method. https://github.com/geopython/pygeoapi/blob/master/pygeoapi/provider/sensorthings.py#L173. Throws an error if you try to use the provider in pygeoapi